skopt/plots.py

```diff
@@ -93,322 +98,880 @@ def plot_convergence(*args, **kwargs):
-                rvs_[:, (j, i)] = (x_, y_)
-                row.append(np.mean(model.predict(rvs_)))
-            zi.append(row)
+        # Search-space boundary for this dimension.
+        bounds_row = dim_row.bounds
 
-        return xi, yi, np.array(zi).T
+        # Map the number of bins to a log-space if necessary.
+        bins_mapped = _map_bins(bins=bins,
+                                bounds=dim_row.bounds,
+                                prior=dim_row.prior)
 
+        # Plot a histogram on the diagonal.
+        ax[row, row].hist(samples_row, bins=bins_mapped, range=bounds_row)
 
-def plot_objective(result, levels=10, n_points=40, n_samples=250, size=2,
-                   zscale='linear', dimensions=None):
-    """Pairwise partial dependence plot of the objective function.
+        # For all columns until the diagonal in the 2-d plot matrix.
+        for col in range(row):
+            # Get the search-space dimension for this column.
+            dim_col = dimensions[col]
 
-    The diagonal shows the partial dependence for dimension `i` with
-    respect to the objective function. The off-diagonal shows the
-    partial dependence for dimensions `i` and `j` with
-    respect to the objective function. The objective function is
-    approximated by `result.model.`
+            # Get the index for this search-space dimension.
+            # This is used to lookup that dimension in some functions.
+            index_col = dim_col.index
 
-    Pairwise scatter plots of the points at which the objective
-    function was directly evaluated are shown on the off-diagonal.
-    A red point indicates the found minimum.
+            # Get the samples from the optimization-log for that dimension.
+            samples_col = get_samples_dimension(result=result, index=index_col)
 
-    Note: search spaces that contain `Categorical` dimensions are
-          currently not supported by this function.
+            # Plot all the parameters that were sampled during optimization.
+            # These are plotted as small coloured dots, where the colour-shade
+            # indicates the time-progression.
+            ax[row, col].scatter(samples_col, samples_row,
+                                 c=sample_order, s=40, lw=0., cmap='viridis')
+
+            # Get the best-found sample for this dimension.
+            best_sample_col = result.x[index_col]
+
+            # Plot the best parameters that were sampled during optimization.
+            # These are plotted as a big red star.
+            ax[row, col].scatter(best_sample_col, best_sample_row,
+                                 c='red', s=100, lw=0., marker='*')
+
+    # Make various adjustments to the plots.
+    _adjust_fig(fig=fig, ax=ax, space=space,
+                dimensions=dimensions, ylabel="Sample Count")
+
+    return fig, ax
+
+
+def plot_objective(result, levels=10, n_points=40, n_samples=250,
+                          zscale='linear', dimension_names=None):
```

Comment:
remove some leading white space to fix the indentation

Reply:
Fixed.